### ISSUE TITLE
Notification not showing full sequence for 3+ level

### ISSUE THREAD

**Issue by ww7** [zenangst/KeyboardCowboy#548]
For 3 levels sequences like `Fn+V,E,P`, bundle notification shows only last key in sequence (`P`).

**Comment by zenangst** [comment]
Yeah, I've noticed this, will investigate and fix this :)

**Comment by zenangst** [comment]
Figure out a fix for this, will be a part of the next version.

**Comment by zenangst** [comment]
@ww7 I did some changes in the latest version to fix this issue, would you mind verifying if this is fixed now?

https://github.com/zenangst/KeyboardCowboy/releases/tag/3.25.0

**Comment by ww7** [comment]
@zenangst text titles in notifications not fit a popup, I think it can be in line under icon and keys
![CleanShot 2024-10-25 at 11 01 41@2x](https://github.com/user-attachments/assets/e255a5ba-f06b-4a80-8666-1b9ef8981adf)

**Comment by ww7** [comment]
Now 3-rd level shows full keyboard sequence.
Only thing about levels, 2-nd level not informing about existing of 3-d.

**Comment by zenangst** [comment]
Ah, I see now.
I think you are right that we need some updates when it comes to the UI.

> Only thing about levels, 2-nd level not informing about existing of 3-d.

My idea here was to:

- First key is pressed
  - Show next level
- Second key is pressed
  - Expand on next
- Final key is pressed
  - Show full sequence

**Comment by ww7** [comment]
Yeah, but would be good to inform if next level available, like show keys and title "next level" or arrow icon

**Comment by zenangst** [comment]
Not a bad suggestion, I'll see if we can't redesign this for the next version.

**Comment by zenangst** [comment]
Oh... and thanks a lot for all the feedback, I really appreciate it 😍

**Comment by zenangst** [comment]
The issue has evolved, so I removed the tags

**Comment by zenangst** [comment]
@ww7 worked a bit on the redesign just now.

<img width="1378" alt="image" src="https://github.com/user-attachments/assets/276b82d8-5882-43d7-bbe1-ba6d60d6ed87">

**Comment by ww7** [comment]
I think align left App icon to Title would be better, as we mostly reading left to right

**Comment by zenangst** [comment]
Made another iteration now

<img width="1422" alt="image" src="https://github.com/user-attachments/assets/e773acf2-9d84-4a66-b4a8-e3d2afde2ed1">

**Comment by ww7** [comment]
Try place icon to left side in row with keys?

**Comment by zenangst** [comment]
Played around with this a bit more.

Now the key that you pressed has less opacity.
The next one in the sequence has a glow.

<img width="300" alt="image" src="https://github.com/user-attachments/assets/10a01b80-dd10-41af-8901-3e0bf14b643a">

**Comment by ww7** [comment]
Why not try to keep Icon with Title? 
![CleanShot 2024-10-27 at 07 07 55](https://github.com/user-attachments/assets/3faa5482-35d6-4028-8253-4685181cdd30)

**Comment by ww7** [comment]
Why not try to keep Icon with Title?
![CleanShot 2024-10-27 at 07 20 59@2x](https://github.com/user-attachments/assets/94d6f43a-9ae6-4f9f-ba56-761d8d621019)

**Comment by ww7** [comment]
If icon now at right. maybe also align title to right?

**Comment by ww7** [comment]
Icons horizontally jumping when loading

**Comment by zenangst** [comment]
@ww7 I had to remove an operator because of a crash, will take another stab at this later.
Currently my focus is on https://github.com/zenangst/KeyboardCowboy/issues/565

**Comment by ww7** [comment]
Recalling issue with notification appearance.
![CleanShot 2024-12-09 at 05 26 35@2x](https://github.com/user-attachments/assets/7f4fde0c-6aa9-4746-8879-64df365c2955)

1. Icon showing only for apps? Would it be possible to show also for files?
2. Notification truncating long sequences. Shift icon can be smaller.
3. Title would be better to be on the same align as the icon (right)

As for me, icon can be smaller and in same line with the title.

**Comment by ww7** [comment]
On such long shortcuts it's still truncating as on screenshot.
And would be lovely to have ability align title right, close to icon.
